diff -Naur a/boost/beast/websocket/impl/ping.hpp b/boost/beast/websocket/impl/ping.hpp
--- a/boost/beast/websocket/impl/ping.hpp	2019-05-06 22:01:43.435117251 -0400
+++ b/boost/beast/websocket/impl/ping.hpp	2019-05-06 22:02:37.949433556 -0400

diff -Naur a/boost/hana/config.hpp b/boost/hana/config.hpp
--- a/boost/hana/config.hpp	2019-04-10 03:35:48.000000000 +0800
+++ b/boost/hana/config.hpp	2023-12-28 20:16:28.522871867 +0800
@@ -89,7 +89,7 @@
 #if defined(_LIBCPP_VERSION)

 #   define BOOST_HANA_CONFIG_LIBCPP BOOST_HANA_CONFIG_VERSION(              \
-                ((_LIBCPP_VERSION) / 1000) % 10, 0, (_LIBCPP_VERSION) % 1000)
+                ((_LIBCPP_VERSION) / 1000), 0, (_LIBCPP_VERSION) % 1000)

 #   if BOOST_HANA_CONFIG_LIBCPP < BOOST_HANA_CONFIG_VERSION(1, 0, 101)
 #       warning "Versions of libc++ prior to the one shipped with Clang 3.5.0 are not supported by Hana."
